app/controllers/game.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Module dependencies.
+ */
+const mongoose = require('mongoose'),
+  Game = mongoose.model('Game');
+
+/**
+ * Save Game
+ */
+exports.saveGame = (req, res) => {
+  if (req.body.gameId) {
+    Game.findOneAndUpdate({ gameId: req.body.gameId })
+      .exec((err, game) => {
+        if (err) return res.status(500).json({ message: 'An error occured' });
+        if (!game) return res.status(404).json({ message: 'Game not found' });
+        game.gameId = req.body.gameId;
+        game.players = req.body.players;
+        game.gameWinner = req.body.gameWinner;
+
+        game.save((err) => {
+          if (err) {
+            return res.status(500).json({ error: err.errors });
+          }
+          return res.status(200).json({ message: 'Game Saved', game });
+        });
+      });
+  } else {
+    return res.status(500).json({ message: 'Internal Server Error' });
```

Comment:
If there's no `gameId` coming from `req.body` you return a `400 bad request` instead of  the`500 ISE`

Reply:
Gotcha